### PR TITLE
implement moderation report handling

### PR DESCRIPTION
**Pull request by Geosearchef**

### CODE REVIEW

**Comment by Rackover** [comment]
https://github.com/FAForever/faf-moderator-client/issues/82

File: src/main/java/com/faforever/moderatorclient/mapstruct/ModerationReportMapper.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import com.faforever.moderatorclient.ui.domain.ModerationReportFX;
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel = "spring", uses = {JavaFXMapper.class, PlayerMapper.class, FeaturedModMapper.class, BanInfoMapper.class, CycleAvoidingMappingContext.class})
+public abstract class ModerationReportMapper {
+	public abstract ModerationReportFX map(ModerationReport dto);
+
+	public abstract ModerationReport map(ModerationReportFX fxBean);
+
+//	public abstract List<ModerationReportFX> mapToFx(List<ModerationReport> dtoList);
```
Comment (axel1200) [inline src/main/java/com/faforever/moderatorclient/mapstruct/ModerationReportMapper.java]: Why?

File: src/main/java/com/faforever/moderatorclient/mapstruct/MapMapper.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 import java.util.List;
 
-@Mapper(uses = {JavaFXMapper.class, MapVersionMapper.class, PlayerMapper.class, CycleAvoidingMappingContext.class})
+@Mapper(componentModel = "spring", uses = {JavaFXMapper.class, MapVersionMapper.class, PlayerMapper.class, CycleAvoidingMappingContext.class})
```
Comment (axel1200) [inline src/main/java/com/faforever/moderatorclient/mapstruct/MapMapper.java]: componentModel = "spring" ... what is that for

File: src/main/java/com/faforever/moderatorclient/ui/MainController.java [hunk 3]
```diff
@@ -69,6 +73,7 @@ public void initializeAfterLogin() {
         initVotingTab();
         initMessagesTab();
         initTutorialTab();
+		initReportTab();
```
Comment (axel1200) [inline src/main/java/com/faforever/moderatorclient/ui/MainController.java]: format?

File: src/main/java/com/faforever/moderatorclient/ui/MainController.java [hunk 4]
```diff
@@ -92,6 +97,13 @@ private void initLadderMapPoolTab() {
         initLoading(ladderMapPoolTab, ladderMapPoolController::refresh);
     }
 
+	private void initReportTab() {
+		moderationReportController = uiService.loadFxml("ui/main_window/report.fxml");
+		reportTab.setContent(moderationReportController.getRoot());
```
Comment (axel1200) [inline src/main/java/com/faforever/moderatorclient/ui/MainController.java]: formatting?

File: src/main/java/com/faforever/moderatorclient/mapstruct/MapMapper.java (hunk 2 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/moderatorclient/mapstruct/MapMapper.java]: telling spring that this is a component

File: src/main/java/com/faforever/moderatorclient/ui/MainController.java (hunk 3 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/moderatorclient/ui/MainController.java]: intellij was set to detect indents, all indents are spaces but it used tabs, hmm, interesting

**Comment by Geosearchef** [comment]
requested changes addressed,
@Brutus5000 updating a report works fine for me (prod server, status, private note and public note)

**Comment by Brutus5000** [comment]
@Geosearchef I tested it against a local API on latest develop.
The error occurs only if no reported account is associated. I don't know if this case can occur with the web-frontend. In the database it is a valid state (1:n relationship).

**Comment by Rackover** [comment]
Cannot occur with web frontend, however I would prefer the API to be fireproof

**Comment by bukajsytlos** [comment]
As far as I remember, there is constraint that there should be at least one reported user. So API is OK. This have to be handled in client.

**Comment by Rackover** [comment]
Webpage handles it so it looks like we're okay both sides. Ready to merge ?

**Pull request merged**